interfaces/m3db/fs.go

```diff
@@ -51,6 +51,9 @@ type FileSetReader interface {
 	// Read returns the next key and data pair or error, will return io.EOF at end of volume
 	Read() (key string, data []byte, err error)
 
+	// Validate validates the data and returns an error if the data are corrupted
```

Comment:
I think either way is fine? https://en.wikipedia.org/wiki/Data_(word)

"In scientific writing data is often treated as a plural...British usage now widely accepts treating data as singular in standard English...The IEEE Computer Society allows usage of data as either a mass noun or plural based on author preference... IEEE in the editorial style manual indicates to always use the plural form"

lol